libsolidity/grammar.txt

```diff
@@ -5,44 +5,50 @@ ContractPart = StateVariableDeclaration | StructDefinition | ModifierDefinition
 
 InheritanceSpecifier = Identifier ( '(' Expression ( ',' Expression )* ')' )?
 StructDefinition = 'struct' Identifier '{'
-                     ( VariableDeclaration (';' VariableDeclaration)* )? '}
-StateVariableDeclaration = TypeName ( 'public' | 'inheritable' | 'private' )? Identifier ';'
+                     ( VariableDeclaration (';' VariableDeclaration)* )? '}'
+StateVariableDeclaration = TypeName ( 'public' | 'internal' | 'private' )? Identifier ';'
 ModifierDefinition = 'modifier' Identifier ParameterList? Block
 FunctionDefinition = 'function' Identifier ParameterList
-					 ( Identifier | 'constant' | 'external' | 'public' | 'inheritable' | 'private' )*
+                     ( Identifier | 'constant' | 'external' | 'public' | 'internal' | 'private' )*
                      ( 'returns' ParameterList )? Block
 
 EnumValue = Identifier
-EnumDefinition = 'enum' '{' EnumValue (',' EnumValue)* '}'
+EnumDefinition = 'enum' Identifier '{' EnumValue? (',' EnumValue)* '}'
 ParameterList = '(' ( VariableDeclaration (',' VariableDeclaration)* )? ')'
 // semantic restriction: mappings and structs (recursively) containing mappings
 // are not allowed in argument lists
 VariableDeclaration = TypeName Identifier
 TypeName = ElementaryTypeName | Identifier | Mapping | ArrayTypeName
 Mapping = 'mapping' '(' ElementaryTypeName '=>' TypeName ')'
-ArrayTypeName = TypeName '[' (Expression)? ']'
+ArrayTypeName = TypeName ('[' Expression? ']')+
```

Comment:
@walter-weinmann `StructDefinition` is fixed in ff9d6e05f2c49af37e36fae65269959eb21cf6b2. I've added `ForStatement` to `Statement` in 447797ad9c5228ca50a4fa5b17ced4aba07e8d65.

Regarding `Block` statement, then I can't see any differences. `solc` does not complain about blocks in blocks, etc, but I haven't tested it thoroughly.